builder/vsphere/common/step_boot_command.go

```diff
@@ -76,9 +76,8 @@ func (s *StepBootCommand) Run(ctx context.Context, state multistep.StateBag) mul
 		ui.Say(fmt.Sprintf("HTTP server is working at http://%v:%v/", ip, port))
 	}
 
+	var keyAlt, keyCtrl, keyShift bool
 	sendCodes := func(code key.Code, down bool) error {
-		var keyAlt, keyCtrl, keyShift bool
-
 		switch code {
 		case key.CodeLeftAlt:
 			keyAlt = down
```

Comment:
So testing with `"<leftShiftOn>a<leftShiftOff><wait1m><bs>"` was a super great idea and with that, I figured out that there's still some code missing in order to make this work. 

`down` will always default to `keyShift` overriding any `leftShiftOn`/`leftShiftOff`. The results for the test was an lowercase `a`. 

Removing the default from the switch and adding the following works 👍 
```
case key.CodeLeftShift:
			keyShift = down
```

Another small thing to do would be moving the logs in https://github.com/hashicorp/packer/blob/master/common/bootcommand/usb_driver.go#L117 to this file before calling `vm.TypeOnKeyboard` with the actual value of shift. Or maybe just remove the shift info from the first log 🤔  pick the one you think is better!

Reply:
Following this logic ctrl and alt should be already working 👍

Reply:
good thing we can test the once case that was the problem :)

Reply:
the current logic is going to type in the special codes. Do we want things like a double shift/ctrl/alt?

Reply:
If we added a return after, then we would lose the capability to send in the shift/ctrl/alt so leaving as is.

Reply:
I think is ok leaving as it is. Returning would still work for the cases we are talking about and any other case I can think, but I don't know all the possibilities here that could be affected by that.